app.go

```diff
@@ -110,15 +111,15 @@ func compileTime() time.Time {
 // Usage, Version and Action.
 func NewApp() *App {
 	return &App{
-		Name:         filepath.Base(os.Args[0]),
-		HelpName:     filepath.Base(os.Args[0]),
-		Usage:        "A new cli application",
-		UsageText:    "",
-		BashComplete: DefaultAppComplete,
-		Action:       helpCommand.Action,
-		Compiled:     compileTime(),
-		Writer:       os.Stdout,
-		InterruptHandlerFunc: func(c *Context) {},
+		Name:             filepath.Base(os.Args[0]),
+		HelpName:         filepath.Base(os.Args[0]),
+		Usage:            "A new cli application",
+		UsageText:        "",
+		BashComplete:     DefaultAppComplete,
+		Action:           helpCommand.Action,
+		Compiled:         compileTime(),
+		Writer:           os.Stdout,
+		InterruptHandler: func(*Context, context.CancelFunc) {},
```

Comment:
I think the default behaviour should cancel the context, and we should have it as a public type available in the library, together with an alternative publicly available exit on interrupt handler.

Reply:
The `ctx.Done()` channel is already publicly available through `cli.Context` which we are going to pass to the `InterruptHandler`. I just wanted to give more control to the user by allowing them to decide whether or not to decide the context.

If we cancel the context ourselves and then call the `InterruptHandler` with `cli.Context` then the `ctx.Done()` channel won't make much sense as the context would already have been cancelled. Waiting for that channel would essentially become a noop.

Reply:
Sure, but I am arguing that default Ctrl+c behaviour in the library should be cancellatiom of the context, which it does not do now.

I am also arguing that some common behaviours should be readily available in the library, so there should be a handled for ExitOnInterrupt and CancelContextOnInterrupt functions readily available in the library.

Reply:
Common behaviours out of the box is intriguing